app/connectors/connectors/rest/rest-swagger-connector/src/main/java/io/syndesis/connector/swagger/SwaggerConnectorComponent.java

```diff
@@ -76,13 +89,24 @@ public void setAuthenticationType(final AuthenticationType authenticationType) {
         this.authenticationType = authenticationType;
     }
 
+    public void setPassword(final String password) {
+        this.password = password;
+    }
+
     public void setSpecification(final String specification) {
         this.specification = specification;
     }
 
+    public void setUsername(final String username) {
+        this.username = username;
+    }
+
     /* default */ void addAuthenticationHeadersTo(final Map<String, Object> headers) {
         if (authenticationType == AuthenticationType.oauth2) {
             headers.put("Authorization", "Bearer " + accessToken);
+        } else if (authenticationType == AuthenticationType.basic) {
+            final String usernameAndPassword = username + ":" + password;
```

Comment:
Do we care if username has `:` in it?

Reply:
Probably, I don't think connector is proper place to make such decisions, it should take the property on face value. Not sure if there is a way to enforce property syntax for configured properties. Created #841

Reply:
This code path could be reached from multiple places so being defensive here makes sense to me as this is creating the basic auth header.

However... from https://tools.ietf.org/html/rfc2617#section-2 colon isn't valid in username so i think we can assume here that it is safe.